From 5a3de108f26e4b0dddadddbe8ccdb1dd5526771f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Wed, 27 Jan 2016 09:00:22 +0100
Subject: [PATCH] Avoid execution outside phpMyAdmin
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is hacky, but avoids path disclossure on direct access to the
scripts.

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 libraries/phpseclib/Crypt/AES.php      | 3 +++
 libraries/phpseclib/Crypt/Rijndael.php | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/libraries/phpseclib/Crypt/AES.php b/libraries/phpseclib/Crypt/AES.php
index 2bb4d5e8f8a..4f15c80d3f5 100644
--- a/libraries/phpseclib/Crypt/AES.php
+++ b/libraries/phpseclib/Crypt/AES.php
@@ -46,6 +46,9 @@
  * @license   http://www.opensource.org/licenses/mit-license.html  MIT License
  * @link      http://phpseclib.sourceforge.net
  */
+if (! defined('PHPMYADMIN')) {
+    exit;
+}
 
 namespace phpseclib\Crypt;
 
diff --git a/libraries/phpseclib/Crypt/Rijndael.php b/libraries/phpseclib/Crypt/Rijndael.php
index e97239321c4..109d5d6cc90 100644
--- a/libraries/phpseclib/Crypt/Rijndael.php
+++ b/libraries/phpseclib/Crypt/Rijndael.php
@@ -51,6 +51,9 @@
  * @license   http://www.opensource.org/licenses/mit-license.html  MIT License
  * @link      http://phpseclib.sourceforge.net
  */
+if (! defined('PHPMYADMIN')) {
+    exit;
+}
 
 namespace phpseclib\Crypt;
 
